Rewrote JSON API to use plain-old DTOs

clacote opened the pull request:
That is old-school and rebarbative work, but "it just works".

cc @javamind.
Fix #391 #351.

javamind commented on app/controllers/api/ApiMembers.java:
```diff
@@ -1,45 +1,113 @@
+        @Override
+        public AbstractMemberDTO visit(Member member) {
+            MemberSimpleDTO dto = new MemberSimpleDTO();
+            initCommon(member, dto);
```
This initCommon should be used in the methods initSimpleXXX

clacote replied:
Fixed in f5785ac

javamind commented:
:+1:

clacote commented:
Also, I fixed DateTime marshalling (previously used in `AbstractSessionDTO`): replaced `DateTime` types to `String`, to avoid obscure default `DateTime` marshalling.

clacote commented:
@javamind: any chance that you could test your mobile app with a local deployment to detect any regression?

javamind commented:
@clacote difficult before tonight because I have a new laptop and I have'nt Play Framework for the moment

javamind commented:
:clap: :clap: 
ok it works

clacote commented:
So merge'n'deploy? I'm legally not allowed to merge my own pull requests. But I can do it :p

clacote commented:
There is nobody now on mix-it.fr: i'm merging and deploying.
